R/commonTTest.R

```diff
@@ -443,3 +444,108 @@ summarySEwithin <- function(data=NULL, measurevar, betweenvars=NULL, withinvars=
   # Combine the un-normed means with the normed results
   merge(datac, ndatac)
 }
+
+.descriptivesPlotsRainCloudFill <- function(dataset, variable, groups = NULL, yLabel, dependency = c("none", "paired", "repeated"), horiz) {
+  # Adapted under the MIT license from:
+  # van Langen, J. (2020). Open-visualizations in R and Python.
+  # https://github.com/jorvlan/open-visualizations
+  #
+  # Permission is hereby granted, free of charge, to any person obtaining a copy
+  # of this software and associated documentation files (the "Software"), to deal
+  # in the Software without restriction, including without limitation the rights
+  # to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+  # copies of the Software, and to permit persons to whom the Software is
+  # furnished to do so, subject to the following conditions:
+  #
+  # The above copyright notice and this permission notice shall be included in all
+  # copies or substantial portions of the Software.
+
+  n <- nrow(dataset)
+  if (dependency == "paired") {
+    y   <- unlist(dataset)
+    grp <- rep(as.vector(variable), c(n,n))
+  } else {
+    y   <- dataset[,.v(variable)]
+    grp <- dataset[,.v(groups)]
+  }
+  x  <- as.numeric(as.factor(grp)) - 1
+  set.seed(12312414)
+  xj <- jitter(x, amount = 0.1)
+  if (horiz) {
+    xb <- x
+    xj <- xj - 0.3
+  } else {
+    xb <- x*0.4 + max(xj) + 0.4
+  }
+
+  pointDf <- data.frame(x = xj, y = y, grp = grp)
+  boxDf   <- data.frame(x = xb, y = y, grp = grp)
```

Comment:
Can you just make a one `data.frame` with more columns? This seems a bit wasteful copying all the values of `y` and `grp` to different data.frames...

Reply:
Good point, fixed that!